docs/samples/kafka/image_transformer/image_transformer.py

```diff
@@ -41,11 +44,15 @@ def __init__(self, name: str, predictor_host: str):
         self.predictor_host = predictor_host
         self._key = None
 
-    def preprocess(self, inputs: Union[Dict, CloudEvent, InferRequest],
-                   headers: Dict[str, str] = None) -> Union[Dict, InferRequest]:
-        if inputs['EventName'] == 's3:ObjectCreated:Put':
-            bucket = inputs['Records'][0]['s3']['bucket']['name']
-            key = inputs['Records'][0]['s3']['object']['key']
+    async def preprocess(self, inputs: Union[Dict, CloudEvent, InferRequest],
+                         headers: Dict[str, str] = None) -> Union[Dict, InferRequest]:
+        logging.info("Received input %s", inputs)
+        event_data = await super().preprocess(inputs, headers) if inspect.iscoroutinefunction(self.preprocess) \
```

Comment:
Depends on https://github.com/kserve/kserve/pull/2881 to remove the necessarity of calling super().preprocess.

Reply:
removed super().preprocess call